core-spi-test/src/main/java/org/ehcache/internal/store/StoreEventListenerTest.java

```diff
@@ -223,9 +256,47 @@ public void testPutOnEviction() throws Exception {
     Map<K, V> map = convertToMap(kvStore.iterator());
     assertThat(map.size(), is(lessThan(2)));
     assertThat(listener.evicted, anyOf(hasItem(k), hasItem(k2)));
+    assertThat(listener.expired, not(anyOf(hasItem(k), hasItem(k2))));
+  }
+
+  @SPITest
+  public void testEventListenerInAbsenceOfEvictionEvent() throws Exception {
+    TestTimeSource timeSource = new TestTimeSource();
+    Store<K, V> kvStore = factory.newStore(factory.newConfiguration(factory.getKeyType(), factory.getValueType(),
+        2L, null, null));
+    TestStoreEventListener listener = new TestStoreEventListener();
+    kvStore.enableStoreEventNotifications(listener);
+    kvStore.put(k, v);
+    kvStore.put(k2, v2);
+    timeSource.advanceTime(1);
```

Comment:
why are you calling advanceTime() here? I fail to see what its usefulness is: isn't the store configured to not expire?